src/Workspaces/Core/Portable/LinkedFileDiffMerging/LinkedFileDiffMergingLogger.cs

```diff
@@ -0,0 +1,152 @@
+﻿using System;
+using System.Collections.Concurrent;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.Internal.Log;
+using static Microsoft.CodeAnalysis.LinkedFileDiffMergingSession;
+
+namespace Microsoft.CodeAnalysis
+{
+    internal class LinkedFileDiffMergingLogger
+    {
+        private const string Id = nameof(Id);
+        private static ConcurrentDictionary<int, LogAggregator> LogAggregators = new ConcurrentDictionary<int, LogAggregator>();
+
+        internal enum MergeInfo
+        {
+            SessionsWithLinkedFiles,
+            LinkedFileGroupsProcessed,
+            IdenticalDiffs,
+            IsolatedDiffs,
+            OverlappingDistinctDiffs,
+            OverlappingDistinctDiffsWithSameSpan,
+            OverlappingDistinctDiffsWithSameSpanAndSubstringRelation,
+            InsertedMergeConflictComments,
+            InsertedMergeConflictCommentsAtAdjustedLocation
+        }
+
+        internal static void LogSession(Workspace workspace, LinkedFileDiffMergingSessionInfo sessionInfo)
+        {
+            if (sessionInfo.LinkedFileGroups.Count > 1)
```

Comment:
Telemetry has shown that this case almost *never* happens
```
if (groupInfo.InsertedMergeConflictComments > 0 ||
   groupInfo.InsertedMergeConflictCommentsAtAdjustedLocation > 0)
```

so its ok to log to AI since we will get about 1 per million sessions if at all.

The other events are very common so we need to aggregate them